tests/nosetests/dracut_tests/test_driver_updates.py

```diff
@@ -84,6 +86,95 @@ def makefiles(self, *paths):
         return [makefile(os.path.normpath(self.tmpdir+'/'+p)) for p in paths]
 
 
+class DebuggingTestCase(FileTestCaseBase):
+    def test_debug_mode(self):
+        """check if debug mode is correctly set"""
+        test_cmdline = os.path.join(self.tmpdir, "empty_cmdline")
+        touch(test_cmdline)
+
+        # test with empty cmdline file
+        self.assertFalse(is_debug_mode_enabled(test_cmdline))
+
+        # test no debug without empty cmdline file
+        test_cmdline = os.path.join(self.tmpdir, "nodebug_cmdline")
+        with open(test_cmdline, "wt") as f:
+            f.write("root=/my/cool/root inst.text rd.break")
+        self.assertFalse(is_debug_mode_enabled(test_cmdline))
+
+        # test with inst.debug
+        test_cmdline = os.path.join(self.tmpdir, "inst_debug_cmdline")
+        with open(test_cmdline, "wt") as f:
+            f.write("root=/my/cool/root inst.debug")
+        self.assertTrue(is_debug_mode_enabled(test_cmdline))
+
+        # test with rd.debug
+        test_cmdline = os.path.join(self.tmpdir, "rd_debug_empty_cmdline")
+        with open(test_cmdline, "wt") as f:
+            f.write("rd.debug root=/my/cool/root")
+        self.assertTrue(is_debug_mode_enabled(test_cmdline))
+
+        # test both inst.debug and rd.debug is there
+        test_cmdline = os.path.join(self.tmpdir, "rd_inst_debug_empty_cmdline")
+        with open(test_cmdline, "wt") as f:
+            f.write("rd.debug root=/my/cool/root inst.debug")
+        self.assertTrue(is_debug_mode_enabled(test_cmdline))
+
+    @mock.patch("driver_updates.log")
+    @mock.patch("driver_updates.is_debug_mode_enabled")
+    def test_setup_log(self, mock_is_debug_mode_enabled, mock_log):
+        """check logging setup when debug mode is not enabled"""
+        # run test with disabled debug mode
+        mock_is_debug_mode_enabled.return_value = False
+        setup_log()
+
+        # check logging level is set to DEBUG
+        mock_log.setLevel.assert_called_once_with(logging.DEBUG)
+
+        calls = mock_log.addHandler.mock_calls
+        calls = list(map(repr, calls))
+
+        # with debug mode disabled
+        # DEBUG logs go to journal only
+        self.assertIn("call(<SysLogHandler (DEBUG)>)", calls)
+        # and INFO logs to the console
+        self.assertNotIn(
+            "call(<StreamHandler <_io.FileIO name=8 mode='rb+' closefd=True> (DEBUG)>)",
```

Comment:
I think it would be better to just use the calls as objects, not converting to strings...

Reply:
I was thinking about that but the issue is that you then comparing two python objects, so you have to deal with identity.

I played with that today and found out:
this call:
```
        self.assertIn(mock.call(syslog), calls)
```
will give you this error:
```
AssertionError: call(<SysLogHandler (DEBUG)>) not found in [call(<StreamHandler <stderr> (INFO)>), call(<SysLogHandler (DEBUG)>)]
```

as you can see `SysLogHandler (DEBUG)` is there but it's not the same object so it's false statement. For that reason I decided to compare with repr which will solve this identity issue for you.

Do you have an idea how to resolve this issue?

Reply:
Ohhhh. I get it now. Yes, you can't compare the objects like that. That's why `Mock` has things like [`assert_any_call`](https://docs.python.org/3/library/unittest.mock.html?highlight=assert_any_call#unittest.mock.Mock.assert_any_call).

Reply:
Yes, but the issue here is that the call argument is the object I want to check. So it will again compare just objects and you need to find out what object type it was.

There is solution, you have to unpack all the calls and use `instanceof` and check logLevel. But given to fact that you are testing one of many is present it is much more code and issues than just comparing output of the repr call.

You would have to do something like.

```
self.assertTrue(any(filter(custom_check_func(map(unpack, calls)))
```
and you have to define tho `custom_check_func` which would basically check `instanceof` and `logLevel`. Honestly, I think using `repr` is better.


Another option when I'm thinking about that. is to mock the whole `SysLogHandler` and similar classes. Benefit is that you can check your mock was created. Downside - more mocking. Still not sure if it is worth the trouble to make this working.

Reply:
I guess the best approach here would be to change the code to abstract away handler creation so I can check the function which creates these objects. That would work well but it's more an upstream work than something to do here. At the end there is not a big chance that the repr output will change on RHEL-8.